File: yjKim513/프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    clothes_categorized = {}
    for item, category in clothes:
        if category not in clothes_categorized.keys():
            clothes_categorized[category] = 2
        else:
            clothes_categorized[category] += 1
    
    answer = 1
    for category in clothes_categorized.keys():
        answer *= clothes_categorized[category]
    
    return answer - 1
    
    

File: yjKim513/프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string, int> clothes_categorized;
    
    for (auto iter = clothes.begin(); iter != clothes.end(); iter++) {
        auto search = clothes_categorized.find((*iter)[1]);
        if (search == clothes_categorized.end())
            clothes_categorized.insert({(*iter)[1], 2});
        else
            (search->second)++;
    }
    
    int answer = 1;
    for (auto iter = clothes_categorized.begin(); iter != clothes_categorized.end(); iter++)
        answer *= iter->second;
    
    return --answer;
}